### PR TITLE
Add ARK CAN GPS Board Config

### PR DESCRIPTION
**Pull request by AlexKlimaj**
This PR adds support for the ARK CAN GPS. 
https://arkelectron.com/product/ark-gps/

SAM_0852 (Large)

This also rolls back the GPS submodule to commit https://github.com/PX4/PX4-GPSDrivers/commit/f82313f082f6a9e63f83040996172906622f01a7 to fix https://github.com/PX4/PX4-GPSDrivers/issues/82

Further improvements to the safety button, buzzer, and safety LED over UAVCAN will be made.

    UAVCAN GPS, Magnetometer, Barometer, IMU, Buzzer, and Safety Switch Module
    Runs PX4 Open Source Firmware
    Supports UAVCAN Firmware Updating
    Sensors
        Ublox M9N GPS
            Ultra-robust meter-level GNSS positioning
            Maximum position availability with concurrent reception of 4 GNSS
            Advanced spoofing and jamming detection
            Excellent RF interference mitigation
        Bosch BMM150 Magnetometer
        Bosch BMP388 Barometer
        Invensense ICM-42688-P 6-Axis IMU
    STM32F412CEU6 MCU
    Safety Button
    Buzzer
    Two Pixhawk Standard CAN Connectors
        4 Pin JST GH
    Pixhawk Standard Debug Connector
        6 Pin JST SH
    Small Form Factor
        5cm x 5cm x 1cm
    LED Indicators
        Safety LED
        GPS Fix
        RGB System Status
    USA Built

Flight Logs on 1.12beta5.
https://review.px4.io/plot_app?log=04a5ca3e-91af-4a25-939d-0f61700a226b
https://review.px4.io/plot_app?log=d1e17f9a-1805-49c3-81b3-08f150506ac3

Shell on boot.
```
ERROR [PX4_MTD] failed to locate spi bus
ERROR [PX4_MTD] mtd failure: -6 bus 2 address 0 class 1
nsh: sercon: command not found
nsh: ver: command not found
nsh: mft: command not found
nsh: mft: command not found
Board architecture defaults: /etc/init.d/rc.board_arch_defaults
ERROR [param] Parameter EKF2_MULTI_IMU not found.
ERROR [param] Parameter SENS_IMU_MODE not found.
ERROR [param] Parameter EKF2_MULTI_MAG not found.
ERROR [param] Parameter SENS_MAG_MODE not found.
Board defaults: /etc/init.d/rc.board_defaults
nsh: rgbled: command not found
nsh: rgbled_ncp5623c: command not found
Board sensors: /etc/init.d/rc.board_sensors
ERROR [parameters] get: param 65535 invalid
ERROR [parameters] get: param 65535 invalid
icm42688p #0 on SPI bus 1 (devid=0x26)
bmp388 #0 on I2C bus 2
bmm150 #0 on I2C bus 1
ERROR [param] Parameter SENS_EN_PX4FLOW not found
INFO  [uavcannode] Node ID 125, bitrate 1000000

NuttShell (NSH) NuttX-10.0.0
nsh>
```

### CODE REVIEW

**Comment by dagar** [comment]
The debug build is overflowing slightly.

Screenshot from 2021-05-16 12-46-35

File: boards/ark/can-gps/debug.cmake [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+		bootloaders
+		gps
+		imu/invensense/icm42688p
+		magnetometer/bosch/bmm150
+		safety_button
+		tone_alarm
+		uavcannode
+	MODULES
+		#ekf2
+		load_mon
```
Comment (dagar) [inline boards/ark/can-gps/debug.cmake:24]:
```suggestion
		#load_mon
```
easy flash fix

**Review by dagar — APPROVED** [review]

**Comment by AlexKlimaj** [comment]
Thanks @dagar !
One other oddity is that when CONSTRAINED_FLASH is enabled the build will fail. Looks like it is coming from the gps driver since it will build when the gps driver is excluded. 

```
Memory region         Used Size  Region Size  %age Used
           flash:      312753 B       448 KB     68.17%
            sram:       12564 B       192 KB      6.39/usr/lib/gcc/arm-none-eabi/9.2.1/../../../arm-none-eabi/bin/ld: /usr/lib/gcc/arm-none-eabi/9.2.1/../../../arm-none-eabi/lib/thumb/v7e-m+fp/hard/libm.a(lib_a-e_rem_pio2.o): in function `__ieee754_rem_pio2':
/build/newlib-CVVEyx/newlib-3.3.0/build/arm-none-eabi/thumb/v7e-m+fp/hard/newlib/libm/math/../../../../../../../../newlib/libm/math/e_rem_pio2.c:174: undefined reference to `__aeabi_d2iz'
/usr/lib/gcc/arm-none-eabi/9.2.1/../../../arm-none-eabi/bin/ld: /build/newlib-CVVEyx/newlib-3.3.0/build/arm-none-eabi/thumb/v7e-m+fp/hard/newlib/libm/math/../../../../../../../../newlib/libm/math/e_rem_pio2.c:174: undefined reference to `__aeabi_d2iz'
/usr/lib/gcc/arm-none-eabi/9.2.1/../../../arm-none-eabi/bin/ld: /build/newlib-CVVEyx/newlib-3.3.0/build/arm-none-eabi/thumb/v7e-m+fp/hard/newlib/libm/math/../../../../../../../../newlib/libm/math/e_rem_pio2.c:129: undefined reference to `__aeabi_d2iz'
/usr/lib/gcc/arm-none-eabi/9.2.1/../../../arm-none-eabi/bin/ld: /usr/lib/gcc/arm-none-eabi/9.2.1/../../../arm-none-eabi/lib/thumb/v7e-m+fp/hard/libm.a(lib_a-k_cos.o): in function `__kernel_cos':
/build/newlib-CVVEyx/newlib-3.3.0/build/arm-none-eabi/thumb/v7e-m+fp/hard/newlib/libm/math/../../../../../../../../newlib/libm/math/k_cos.c:78: undefined reference to `__aeabi_d2iz'
/usr/lib/gcc/arm-none-eabi/9.2.1/../../../arm-none-eabi/bin/ld: /usr/lib/gcc/arm-none-eabi/9.2.1/../../../arm-none-eabi/lib/thumb/v7e-m+fp/hard/libm.a(lib_a-k_rem_pio2.o): in function `__kernel_rem_pio2':
/build/newlib-CVVEyx/newlib-3.3.0/build/arm-none-eabi/thumb/v7e-m+fp/hard/newlib/libm/math/../../../../../../../../newlib/libm/math/k_rem_pio2.c:198: undefined reference to `__aeabi_d2iz'
/usr/lib/gcc/arm-none-eabi/9.2.1/../../../arm-none-eabi/bin/ld: /usr/lib/gcc/arm-none-eabi/9.2.1/../../../arm-none-eabi/lib/thumb/v7e-m+fp/hard/libm.a(lib_a-k_rem_pio2.o):/build/newlib-CVVEyx/newlib-3.3.0/build/arm-none-eabi/thumb/v7e-m+fp/hard/newlib/libm/math/../../../../../../../../newlib/libm/math/k_rem_pio2.c:199: more undefined references to `__aeabi_d2iz' follow
collect2: error: ld returned 1 exit status
%
ninja: build stopped: subcommand failed.
make: *** [Makefile:228: ark_can-gps] Error 1
```

**Comment by JacobCrabill** [comment]
> Thanks @dagar ! One other oddity is that when CONSTRAINED_FLASH is enabled the build will fail. Looks like it is coming from the gps driver since it will build when the gps driver is excluded.
> 
> ```
> Memory region         Used Size  Region Size  %age Used
>            flash:      312753 B       448 KB     68.17%
>             sram:       12564 B       192 KB      6.39/usr/lib/gcc/arm-none-eabi/9.2.1/../../../arm-none-eabi/bin/ld: /usr/lib/gcc/arm-none-eabi/9.2.1/../../../arm-none-eabi/lib/thumb/v7e-m+fp/hard/libm.a(lib_a-e_rem_pio2.o): in function `__ieee754_rem_pio2':
> /build/newlib-CVVEyx/newlib-3.3.0/build/arm-none-eabi/thumb/v7e-m+fp/hard/newlib/libm/math/../../../../../../../../newlib/libm/math/e_rem_pio2.c:174: undefined reference to `__aeabi_d2iz'
> /usr/lib/gcc/arm-none-eabi/9.2.1/../../../arm-none-eabi/bin/ld: /build/newlib-CVVEyx/newlib-3.3.0/build/arm-none-eabi/thumb/v7e-m+fp/hard/newlib/libm/math/../../../../../../../../newlib/libm/math/e_rem_pio2.c:174: undefined reference to `__aeabi_d2iz'
> /usr/lib/gcc/arm-none-eabi/9.2.1/../../../arm-none-eabi/bin/ld: /build/newlib-CVVEyx/newlib-3.3.0/build/arm-none-eabi/thumb/v7e-m+fp/hard/newlib/libm/math/../../../../../../../../newlib/libm/math/e_rem_pio2.c:129: undefined reference to `__aeabi_d2iz'
> /usr/lib/gcc/arm-none-eabi/9.2.1/../../../arm-none-eabi/bin/ld: /usr/lib/gcc/arm-none-eabi/9.2.1/../../../arm-none-eabi/lib/thumb/v7e-m+fp/hard/libm.a(lib_a-k_cos.o): in function `__kernel_cos':
> /build/newlib-CVVEyx/newlib-3.3.0/build/arm-none-eabi/thumb/v7e-m+fp/hard/newlib/libm/math/../../../../../../../../newlib/libm/math/k_cos.c:78: undefined reference to `__aeabi_d2iz'
> /usr/lib/gcc/arm-none-eabi/9.2.1/../../../arm-none-eabi/bin/ld: /usr/lib/gcc/arm-none-eabi/9.2.1/../../../arm-none-eabi/lib/thumb/v7e-m+fp/hard/libm.a(lib_a-k_rem_pio2.o): in function `__kernel_rem_pio2':
> /build/newlib-CVVEyx/newlib-3.3.0/build/arm-none-eabi/thumb/v7e-m+fp/hard/newlib/libm/math/../../../../../../../../newlib/libm/math/k_rem_pio2.c:198: undefined reference to `__aeabi_d2iz'
> /usr/lib/gcc/arm-none-eabi/9.2.1/../../../arm-none-eabi/bin/ld: /usr/lib/gcc/arm-none-eabi/9.2.1/../../../arm-none-eabi/lib/thumb/v7e-m+fp/hard/libm.a(lib_a-k_rem_pio2.o):/build/newlib-CVVEyx/newlib-3.3.0/build/arm-none-eabi/thumb/v7e-m+fp/hard/newlib/libm/math/../../../../../../../../newlib/libm/math/k_rem_pio2.c:199: more undefined references to `__aeabi_d2iz' follow
> collect2: error: ld returned 1 exit status
> %
> ninja: build stopped: subcommand failed.
> make: *** [Makefile:228: ark_can-gps] Error 1
> ```

This seems to have been fixed at some point; @dagar do you have any idea what may have fixed this?  Trying to apply some changes to Ark GPS support on a private fork and don't know how to tackle this one.  It seems to be related to the link order of the libraries `-lm` and `-lgcc`, but I don't know how to get CMake to use a specific link order for external libraries.

**Pull request merged**